Formula/lib/libsoup@2.rb

```diff
@@ -22,16 +22,24 @@ class LibsoupAT2 < Formula
   depends_on "gobject-introspection" => :build
   depends_on "meson" => :build
   depends_on "ninja" => :build
-  depends_on "pkg-config" => :build
+  depends_on "pkg-config" => [:build, :test]
   depends_on "python@3.12" => :build
   depends_on "vala" => :build
+
+  depends_on "glib"
   depends_on "glib-networking"
   depends_on "gnutls"
+  depends_on "icu4c"
```

Comment:
I don't see linkage with `icu4c`?

Reply:
icu4c is required by libpsl (so I feel like it is better to add icu4c as dep?)

```
==> pkg-config --cflags --libs libsoup-2.4
Package icu-uc was not found in the pkg-config search path.
Perhaps you should add the directory containing `icu-uc.pc'
to the PKG_CONFIG_PATH environment variable
Package 'icu-uc', required by 'libpsl', not found
```

Reply:
No linkage though, so this seems like an indirect dependency that doesn't need to be declared here.

It may be better to modify `libpsl`'s `.pc` file so that it can find `icu4c` without having to modify `PKG_CONFIG_PATH`.

Reply:
that works for me.